tests/integration/delete_user.t

```diff
@@ -0,0 +1,123 @@
+#!/usr/bin/perl -w
+
+use ProductOpener::PerlStandards;
+
+use Test::More;
+use ProductOpener::APITest qw/:all/;
+use ProductOpener::Test qw/:all/;
+use ProductOpener::TestDefaults qw/:all/;
+use ProductOpener::Users qw/:all/;
+use ProductOpener::Producers qw/:all/;
+
+remove_all_users();
+wait_application_ready();
+
+#new common user agent
+my $ua = new_client();
+my %create_client_args = (%default_user_form, (email => 'bob@test.com'));
+create_user($ua, \%create_client_args);
+
+#new admin user agent
+my $admin = new_client();
+create_user($admin, \%admin_user_form);
+
+#common ua add a new product then delete the account while being still logged in
+edit_product($ua, \%default_product);
+
+my @words = (
+	"Delete the user",
+	"User is being deleted. This may take a few minutes.",
+	"See you soon!",
+	"Invalid user.",
+	"Unknown user.",
+	"Incorrect user name or password.",
+	"/editor/anonymous"
+);
+my $url_userid = construct_test_url("/cgi/user.pl?type=edit&userid=tests", "world");
+my $url_delete = construct_test_url("/cgi/user.pl", "world");
+my $response_edit = $ua->get($url_userid);
+
+my %delete_form = (
+	name => 'Test',
+	email => 'bob@test.com',
+	password => '',
+	confirm_password => '',
+	delete => 'on',
+	action => 'process',
+	type => 'edit',
+	userid => 'tests'
+);
+
+#checking if the delete button exist
+like($response_edit->content, qr/\Q$words[0]\E/i, "the delete button does exist");
+
+#deleting the account
+my $before_delete_ts = time();
+my $response_delete = $ua->post($url_delete, \%delete_form);
+#checking if we are redirected to the account deleted page
+like($response_delete->content, qr/\Q$words[1]\E/i, "the account was deleted");
+
+#waiting the deletion task to be done
+my $jobs = $minion->jobs({tasks => ["delete_user_task"]});
+#iterate on job
+while (my $job = $jobs->next) {
+	#only those who were created after the timestamp
+	my $waited = 0;    #waiting time
+	if ($job->created > $before_delete_ts) {
+		#waiting the job to be done
+		while ($job->state == "inactive" or $job->state == "active" or $waited < 200) {
+			sleep(2);
+			$waited++;
+		}
+	}
+}
+#checking if there is still delete_user_task jobs
+$jobs = $minion->jobs({tasks => ["delete_user_task"]});
+my $jobs_count = 0;
+while (my $job = $jobs->next) {
+	$jobs_count++;
+}
```

Comment:
I would like you to put the logic of returning jobs corresponding to a task and started after a given ts in TestAPI.pm (because we might reuse it).

so this code would read:
```perl
my $jobs_ref = get_minion_jobs("delete_user_task", $before_delete_ts, 400);
is(scalar @{$jobs_ref}, 1, "We have one job launched to remove user"); 
is($jobs_ref->[0]->state, "finished", "And job is completed");
```
get_minion_jobs would return all job of type delete_user_task (first argument, eg $job_name)  started after $before_delete_ts (second argument, eg $after_ts) waiting for each job at most 400 seconds ($max_wait).

This way it will test that the state is "finished" and not "failed".
Because otherwise it's not easy to understand why following tests will fail, while here we got a good indicator.

Reply:
I'm currently working on it, but I have a "big" trouble : when importing ProductOpener::Producers into APITest.pm, whatever test I run it encounters "Error 137", I don't know what to do (I think it may be a compilation error but why ?)